WordPress/Classes/Services/PostCoordinator.swift

```diff
@@ -71,15 +71,25 @@ class PostCoordinator: NSObject {
         return isPushingAllPendingMedia
     }
 
-    func save(_ postToSave: AbstractPost, automatedRetry: Bool = false) {
-        prepareToSave(postToSave, automatedRetry: automatedRetry) { post in
-            self.upload(post: post)
+    func save(_ postToSave: AbstractPost, automatedRetry: Bool = false, completion: ((Result<AbstractPost>) -> ())? = nil) {
+        prepareToSave(postToSave, automatedRetry: automatedRetry) { result in
+            switch result {
+            case .success(let post):
+                self.upload(post: post, completion: completion)
+            case .error(let error):
+                completion?(.error(error))
+            }
         }
     }
 
     func autoSave(_ postToSave: AbstractPost, automatedRetry: Bool = false) {
-        prepareToSave(postToSave, automatedRetry: automatedRetry) { post in
-            self.mainService.autoSave(post, success: { uploadedPost, _ in }, failure: { _ in })
+        prepareToSave(postToSave, automatedRetry: automatedRetry) { result in
+            switch result {
+            case .success(let post):
+                self.mainService.autoSave(post, success: { uploadedPost, _ in }, failure: { _ in })
+            case .error:
+                break
```

Comment:
Should this be "return"? Why are we not doing anything with the error?

Reply:
In this case, `break` or `return` will produce the same result. Do you prefer `return`?

We're not handling the error because the `autoSave` operation is silent. If it fails, we don't show anything to the user. I just kept this behavior.